common/src/main/java/com/wynntils/features/ItemStatInfoFeature.java

```diff
@@ -0,0 +1,593 @@
+
+            boolean isInverted = IdentificationOrderer.INSTANCE.isInverted(idName);
+
+            MutableComponent loreLine = new TextComponent("");
+            loreLine.setStyle(Style.EMPTY.withItalic(false));
+
+            MutableComponent statInfo =
+                    new TextComponent(
+                            (statValue > 0 ? "+" : "") + statValue + type.getInGame(idName));
+            statInfo.setStyle(
+                    Style.EMPTY.withColor(
+                            isInverted ^ (statValue > 0)
+                                    ? ChatFormatting.GREEN
+                                    : ChatFormatting.RED));
+
+            loreLine.append(statInfo);
+
+            String longName = IdentificationContainer.getAsLongName(idName);
+
+            // FIXME: Use actual class spell names (needs current class detection implementation)
+            SpellType spell = SpellType.fromName(longName);
+            if (spell != null) {
+                longName = spell.forOtherClass(ClassType.None).getName() + " Spell Cost";
+            }
+
+            int starsCount = stars.getInt(idName);
+
+            // Add stars
+            if (showStars && stars.contains(idName))
+                loreLine.append(
+                        new TextComponent("***".substring(3 - starsCount))
+                                .withStyle(ChatFormatting.DARK_GREEN));
+
+            loreLine.append(new TextComponent(" " + longName).withStyle(ChatFormatting.GRAY));
+
+            if (idContainer == null || idContainer.isInvalidValue(statValue)) { // id not in api
+                loreLine.append(new TextComponent(" [NEW]").withStyle(ChatFormatting.GOLD));
+                idMap.put(idName, ItemUtils.toLoreStringTag(loreLine));
+                hasNew = true;
+                continue;
+            }
+
+            if (idContainer.hasConstantValue()) {
+                idMap.put(idName, ItemUtils.toLoreStringTag(loreLine));
+                continue;
+            }
+
+            loreLine.append(" ");
+
+            int min = idContainer.getMin();
+            int max = idContainer.getMax();
+            float percentage;
+
+            if (isInverted) {
+                percentage = MathUtils.inverseLerp(max, min, statValue) * 100;
+            } else {
+                percentage = MathUtils.inverseLerp(min, max, statValue) * 100;
+            }
+
+            String loreFormat = alternativeForm ? ALTERNATIVE_FORMAT_STRING : MAIN_FORMAT_STRING;
+
+            Map<String, Component> infoVariables = new HashMap<>();
+
+            IdentificationContainer.ReidentificationChances chances =
+                    idContainer.getChances(statValue, isInverted, starsCount);
+
+            infoVariables.put("percentage", getPercentageTextComponent(percentage));
+
+            // TODO add escaped chars like old impl for special chars
+            infoVariables.put(
+                    "chance_perfect",
```

Comment:
Don't add the unicode symbol or color to the info variable itself, make it optional for the user

Reply:
If we don't add it there, it will not use the correct color. I think we will have to worry about this when we add configs,

Reply:
Can't you put a normal color code before it? Pretty sure that overrides

Reply:
![image](https://user-images.githubusercontent.com/49001742/155589005-b1fae50b-fb2a-4231-99df-3480d219dbfb.png)

Code: 
```java
            infoVariables.put(
                    "chance_perfect",
                    new TextComponent(
                                    String.format(
                                            "\u2605%.2f%%", idContainer.getPerfectChance() * 100)));
            infoVariables.put(
                    "chance_increase",
                    new TextComponent(String.format("\u21E7%.1f%%", chances.increase() * 100)));
            infoVariables.put(
                    "chance_decrease",
                    new TextComponent(String.format("\u21E9%.1f%%", chances.decrease() * 100)));
```
and
```java
    private static final String ALTERNATIVE_FORMAT_STRING =
            "{percentage} {chance_perfect}"
                    + " {chance_increase} {chance_decrease}"
                    + ChatFormatting.YELLOW
                    + " ["
                    + "{min}"
                    + ChatFormatting.YELLOW
                    + ", "
                    + "{max}"
                    + ChatFormatting.YELLOW
                    + "]"; // Used when user presses SHIFT on lore.
```

This confirms that we have to apply the style there.

Min/Max works because it applies the style:
```java
            infoVariables.put(
                    "min",
                    new TextComponent(String.valueOf(min))
                            .withStyle(
                                    Style.EMPTY
                                            .withColor(ChatFormatting.YELLOW)
                                            .withItalic(false)));
            infoVariables.put(
                    "max",
                    new TextComponent(String.valueOf(max))
                            .withStyle(
                                    Style.EMPTY
                                            .withColor(ChatFormatting.YELLOW)
                                            .withItalic(false)));
```

Reply:
Makes sense but it is rather unhappy to force the colors

Reply:
I agree, but it should be fine for now. We can get back to this when formatters aren't hardcoded.